src/Passwordless/Models/AuthenticationOptions.cs

```diff
@@ -1,3 +1,12 @@
+using System;
+
 namespace Passwordless.Models;
 
-public record AuthenticationOptions(string UserId);
\ No newline at end of file
+/// <summary>
+/// Used to manually generate an authentication token.
+/// </summary>
+/// <param name="UserId">User identifier the token is intended for.</param>
+/// <param name="TimeToLive">Optional. How long a token is valid for. Default value is 15 minutes.</param>
+public record AuthenticationOptions(string UserId, TimeSpan? TimeToLive = null);
+
```

Comment:
Why two records?

Reply:
Having two records allows for the user of the sdk to interact with a nullable TimeSpan, which is an easier way to declare a period of time.

The API uses an integer as the ttl. And since source generation is used with the JSON serializer we need a concrete model, hence the internal record used directly for serialization.

Reply:
I'd personally prefer a custom converter than another model, to avoid copying all the other values.